db: fix overflow on systems where time_t and off_t aren't 64bit
https://github.com/cboxdoerfer/fsearch/commit/e76235884d034881948026ce3febabb4bf4cf61f

Note: The line "time_t mtime = 0;" isn't yet present in the code of search_database.c of v0.2.3, so this line of the commit change is ommitted from this patch.


--- a/src/fsearch_database.c
+++ b/src/fsearch_database.c
@@ -290,10 +290,10 @@
 
     if ((index_flags & DATABASE_INDEX_FLAG_SIZE) != 0) {
         // size: size of file/folder
-        off_t size = 0;
+        int64_t size = 0;
         data_block = copy_bytes_and_return_new_src(&size, data_block, 8);
 
-        db_entry_set_size(entry, size);
+        db_entry_set_size(entry, (off_t)size);
     }
 
     if ((index_flags & DATABASE_INDEX_FLAG_MODIFICATION_TIME) != 0) {
@@ -301,7 +301,7 @@
         int64_t mtime = 0;
         data_block = copy_bytes_and_return_new_src(&mtime, data_block, 8);
 
-        db_entry_set_mtime(entry, mtime);
+        db_entry_set_mtime(entry, (time_t)mtime);
     }
 
     return data_block;
